serpentTools/parsers/results.py

```diff
@@ -1,15 +1,247 @@
+        and thus is stored only once. e.g., 'version': 'Serpent 2.1.29'
+    resdata: dict
+        Dictionary with serpent time-dependent variables as keys and the
+        corresponding values. The data is unique for each burnup step.
+        Some variables also contain uncertainties.
+        e.g., 'absKeff': [[9.91938E-01, 0.00145],[1.81729E-01, 0.00240]]
+    univdata: dict
+        Dictionary of universe names and their corresponding
+        :py:class:`~serpentTools.objects.containers.HomogUniv`
+        objects. The keys describe a unique state:
+        'universe', burnup (MWd/kg), burnup index, time (days)
+        ('0', 0.0, 1, 0.0)
+
     Parameters
     ----------
     filePath: str
-        path to the depletion file
+        path to the results file
+
+    Raises
+    ------
+    SerpentToolsException:  Serpent version is not supported
+                            No universes are found in the file
+                            No results are collected
+                            Corrupted results
+    IOError: file is unexpectedly closes while reading
     """
-    pass
+
+    def __init__(self, filePath):
+        XSReader.__init__(self, filePath, 'xs')
+        self.__fileObj = None
+        self.__serpentVersion = rc['serpentVersion']
+        self._counter = {'meta': 0, 'rslt': 0, 'univ': 0}
+        self._univlist = []
+        self.metadata,  self.resdata, self.univdata = {}, {}, {}
+        self._regexpStr = re.compile(r'\'.+\'')  # string
+        self._regexpVec = re.compile(r'(?<==.)\[.+?\]')  # vector
+        self._regexpScl = re.compile(r'=.+;')  # scalar
+        self._strtok = re.compile(r'^\w+')  # first word in the line
+
+    def _read(self):
+        """Read through the results file and store requested data."""
+        info('Preparing to read {}'.format(self.filePath))
+        with open(self.filePath, 'r') as fObject:
+            for tline in fObject:
+                self._processResults(tline)
+        info('Done reading results file')
+
+    def _processResults(self, tline):
+        """Performs the main processing of the results."""
+        #  Not an empty line or a comment
+        if tline.strip() and '%' not in tline and '=' in tline and 'idx = ' not in tline:
+            self._whereAmI(tline)  # identify position in file
+            varNameSer, varNamePy= self._getVarName(tline)  # obtain the name of the variable
+            # should this variable be stored
+            if not self._checkAddVariable(varNameSer):
+                return
+            varType, varVals = self._getVarValues(tline)  # values to be stored
+            if self._posFile == 'meta':
+                self._storeMetaData(varNamePy, varType, varVals)
+            if self._posFile == 'rslt':
+                self._storeResData(varNamePy, varVals)
+            if self._posFile == 'univ':
+                self._storeUnivData(varNameSer, varVals)
+
+    def _whereAmI(self, tline):
+        """Identify the position in the file."""
```

Comment:
posFile is stored on the object simply because the method will not update the position, unless one of the keywords is found. This will lead to returning a non-existing variable.

Reply:
You could have the method return `None` if none of the keywords are found. 

What about the case where `_whereAmI` does not update `posFile`? That would indicate we are not at the beginning of a results, metadata, nor universe block, correct?

Reply:
The posFile will always be one of the these blocks: results, metadata, universes. 
when I scroll through the file and find one of keywords, I update the posFile to its location. 
Example:
if 'VERSION' is found I know that this is the metadata block and thus self._posFile will store this. 
The next variable is still in the results block, however, the posFile will not be updated. Therefore, the function whereAmI will not change the posFile field. 
The posFile will be changed once the results block is found.

Reply:
Ah, got it. Then we do need to keep this as an attribute. Thanks for explaining!